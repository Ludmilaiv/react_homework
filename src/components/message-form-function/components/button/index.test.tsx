import { Button } from './';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

describe('Button', () => {
  it('render component', () => {
    render(<Button className='test' content='click'/>);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Button className='test' content='click'/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with text', () => {
    render(<Button className='test' content='click'/>);
    expect(screen.getByText(/click/)).toBeInTheDocument();
  });

  it('render multiply components', () => {
    render(<>
      <Button className='test' content='click'/>
      <Button className='test' content='click1'/>
      <Button className='test' content='click2'/>
      <Button className='test' content='click3'/>
    </>
    );

    expect(screen.queryAllByRole('button').length).toBe(4);
  });

  it('button is disabled', () => {
    render(<Button className='test' disabled content='click'/>);
    expect(screen.getByRole('button')).toBeDisabled();
  });

  it('button have style border-radius: 3', () => {
    render(<Button className='test' content='click'/>);
    expect(screen.getByRole('button')).toHaveStyle({borderRadius: '3'});
  });

  it('button click with userEvent', async () => {
    const mockHandler = jest.fn();
    render(<Button className='test' onClick={mockHandler} content='click'/>);
    await userEvent.click(screen.getByRole('button'));
    expect(mockHandler).toHaveBeenCalledTimes(1);
  });

  it('button async click', async () => {
    const mockHandler = jest.fn();
    render(<Button className='test' content='click' onClick={() => setTimeout(mockHandler, 2000)} />);
    await userEvent.click(screen.getByText(/click/));
    setTimeout(() => expect(mockHandler).toHaveBeenCalledTimes(1), 2100);
  });
});