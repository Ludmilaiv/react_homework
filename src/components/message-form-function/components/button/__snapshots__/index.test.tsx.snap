// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="test"
    style="border-radius: 3;"
  >
    click
  </button>
</DocumentFragment>
`;
